File: Python/3-Recursion-Programs/48-hcf-using-recursion.py
num1 = int(input("Enter first number: "))
num2 = int(input("Enter second number: "))

def HCF(num1, num2):
    if num2==0:
        return num1
    
    return HCF(num2, num1%num2)

print("\nHCF is:", HCF(num1, num2))

File: Java/3-Recursion-Programs/48-hcf-using-recursion.java
import java.util.Scanner;

class Main {
    static int HCF(int num1, int num2){
        if (num2 == 0) {
            return num1;
        }
        return HCF(num2, num1%num2);
    }

    public static void main(String[] args) {
        Scanner getInput = new Scanner(System.in);

        System.out.println("Enter first number:");
        int num1 = getInput.nextInt();

        System.out.println("Enter second number:");
        int num2 = getInput.nextInt();

        System.out.println("\nHCF is: " + HCF(num1, num2));
        getInput.close();
    }
}